api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java

```diff
@@ -84,35 +200,19 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 		if (birthyear == null) {
 			birthyear = 0;
 		}
-
+		
 		name = name.replaceAll("  ", " ");
 		name = name.replace(", ", " ");
 		String[] names = name.split(" ");
+		Set<Person> result;
 		
 		StringBuilder q = new StringBuilder(
 		        "select p from Person p left join p.names as pname where p.personVoided = false and pname.voided = false and ");
 		
 		if (names.length == 1) {
-			q.append("(").append(" soundex(pname.givenName) = soundex(:n1)").append(
-			    " or soundex(pname.middleName) = soundex(:n1)").append(" or soundex(pname.familyName) = soundex(:n1) ")
-			        .append(" or soundex(pname.familyName2) = soundex(:n1) ").append(")");
+			return  executeSoundexOnePersonNameQuery(name, birthyear, false, gender);
 		} else if (names.length == 2) {
-			q.append("(").append(" case").append("  when pname.givenName is null then 1").append(
-			    "  when pname.givenName = '' then 1").append("  when soundex(pname.givenName) = soundex(:n1) then 4")
-			        .append("  when soundex(pname.givenName) = soundex(:n2) then 3").append("  else 0 ").append(" end")
-			        .append(" + ").append(" case").append("  when pname.middleName is null then 1").append(
-			            "  when pname.middleName = '' then 1").append(
-			            "  when soundex(pname.middleName) = soundex(:n1) then 3").append(
-			            "  when soundex(pname.middleName) = soundex(:n2) then 4").append("  else 0 ").append(" end").append(
-			            " + ").append(" case").append("  when pname.familyName is null then 1").append(
-			            "  when pname.familyName = '' then 1").append(
-			            "  when soundex(pname.familyName) = soundex(:n1) then 3").append(
-			            "  when soundex(pname.familyName) = soundex(:n2) then 4").append("  else 0 ").append(" end").append(
-			            " +").append(" case").append("  when pname.familyName2 is null then 1").append(
-			            "  when pname.familyName2 = '' then 1").append(
-			            "  when soundex(pname.familyName2) = soundex(:n1) then 3").append(
-			            "  when soundex(pname.familyName2) = soundex(:n2) then 4").append("  else 0 ").append(" end")
-			        .append(") > 6");
+			return executeSoundexTwoPersonNameQuery(names[0], names[1], birthyear, false, gender);
 		} else if (names.length == 3) {
```

Comment:
Is there a reason this case isn't covered?

Reply:
I want to split the issue in several parts to make discussion more structured and not make a to large PR. That is why I created a accoding subtask to the main issue.

For now I focused on the case of a names.length of 1 or 2. So we can first agree that this part works. And once this PR is done I will focus on coding this case.

Reply:
That's a very reasonable route to take. Just wanted to understand what is going on here.

Reply:
👍 
There seems like a lot is going on while re-factoring since Lucene core functionality is touched (LuceneQuery.java) which may affects other PersonName searches although the test cases do not indicate it.

And the reverse engineering of the business logic of the sql code is not too trivial: https://talk.openmrs.org/t/soundex-search-in-lucenequeries/26068/22?u=fruether

The other comments should be fixed for now